// @histoire/app/dist/bundled/components/tree/StoryGroup.vue2.js
import { createElementVNode, defineComponent, computed, openBlock, createElementBlock, Fragment, withKeys, createVNode, unref, toDisplayString, createCommentVNode, renderList, createBlock } from "@histoire/vendors/vue";
import StoryListItem from "./StoryListItem.vue.js";
import StoryListFolder from "./StoryListFolder.vue.js";
import { Icon } from "@histoire/vendors/iconify";
import { useFolderStore } from "../../stores/folder.js";
"use strict";
const _hoisted_1 = {
  "data-test-id": "story-group",
  class: "histoire-story-group htw-my-2 first:htw-mt-0 last:htw-mb-0 htw-group"
};
const _hoisted_2 = /* @__PURE__ */ createElementVNode("div", { class: "htw-h-[1px] htw-bg-gray-500/10 htw-mx-6 htw-mb-2 group-first:htw-hidden" }, null, -1);
const _hoisted_3 = ["onKeyup"];
const _hoisted_4 = { class: "htw-truncate" };
const _hoisted_5 = { key: 1 };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryGroup",
  props: {
    path: { default: () => [] },
    group: null,
    stories: null
  },
  setup(__props) {
    const props = __props;
    const folderStore = useFolderStore();
    const folderPath = computed(() => [...props.path, props.group.title]);
    const isFolderOpen = computed(() => folderStore.isFolderOpened(folderPath.value, true));
    function toggleOpen() {
      folderStore.toggleFolder(folderPath.value, false);
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        __props.group.title ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
          _hoisted_2,
          createElementVNode("div", {
            role: "button",
            tabindex: "0",
            class: "htw-px-0.5 htw-py-2 md:htw-py-1.5 htw-mx-1 htw-rounded-sm hover:htw-bg-primary-100 dark:hover:htw-bg-primary-900 htw-cursor-pointer htw-select-none htw-flex htw-items-center htw-gap-2 htw-min-w-0 htw-opacity-50 hover:htw-opacity-100",
            onClick: toggleOpen,
            onKeyup: [
              withKeys(toggleOpen, ["enter"]),
              withKeys(toggleOpen, ["space"])
            ]
          }, [
            createVNode(unref(Icon), {
              icon: unref(isFolderOpen) ? "ri:subtract-line" : "ri:add-line",
              class: "htw-w-4 htw-h-4 htw-ml-4 htw-rounded-sm htw-border htw-border-gray-500/40"
            }, null, 8, ["icon"]),
            createElementVNode("span", _hoisted_4, toDisplayString(__props.group.title), 1)
          ], 40, _hoisted_3)
        ], 64)) : createCommentVNode("", true),
        unref(isFolderOpen) ? (openBlock(), createElementBlock("div", _hoisted_5, [
          (openBlock(true), createElementBlock(Fragment, null, renderList(__props.group.children, (element) => {
            return openBlock(), createElementBlock(Fragment, {
              key: element.title
            }, [
              element.children ? (openBlock(), createBlock(StoryListFolder, {
                key: 0,
                path: unref(folderPath),
                folder: element,
                stories: __props.stories,
                depth: 0
              }, null, 8, ["path", "folder", "stories"])) : (openBlock(), createBlock(StoryListItem, {
                key: 1,
                story: __props.stories[element.index],
                depth: 0
              }, null, 8, ["story"]))
            ], 64);
          }), 128))
        ])) : createCommentVNode("", true)
      ]);
    };
  }
});
export {
  _sfc_main as default
};
